Source/APIClient/Request.swift

```diff
@@ -18,32 +18,18 @@ public enum HTTPMethod: String {
 public protocol Request {
     static var baseURL: String { get }
     static var acceptHeader: String? { get }
-    static var authToken: String? { get set }
+    static var authToken: String? { get }
```

Comment:
But if an auth token is returned from some endpoint, wouldn't we be making that request outside of our `Request` type, in which case we need to set it _on_ `MyRequest` from the outside? So for example, in `SomeViewModel`:
```swift
func authenticateClient() {
        myClient.authenticate { result in
            switch result {
            case .success(let data):
                let code = extractAuthCode(fromData: data))
                self.getToken(authCode: code)
            case .failure(let error):
                print(error)
            }
        }
    }

func getToken(authCode: String) {
        myClient.getToken(code: code) { result in
            switch result {
            case .success(let data):
                let contents = try? JSONSerialization.jsonObject(with: data!) as! [String: Any]
                MyRequest.authToken = contents!["access_token"] as? String                
            case .failure(let error):
                print(error)
            }
        }
    }

```

At least this is how it works for the API I'm playing with now.